### PR TITLE
Skip slow test configurations for slow tests

### PR DESCRIPTION
**Pull request by mpharrigan**
xref #114 

This is about 47 seconds worth of tests

### CODE REVIEW

**Comment by tanujkhattar** [comment]
To be sure, these tests can still be run if we provide a flag like `--enable-slow` ? 

There's a parallel effort happening in Cirq to solve this problem where we can mark tests as "slow" and they'll be run only as part of daily / weekly CI instead of running them on every PR. 

See https://github.com/quantumlib/Cirq/issues/6211 and https://github.com/quantumlib/Cirq/pull/6328

**Comment by mpharrigan** [comment]
This PR doesn't respond to flags; it just skips one of the configurations. You can run it by commenting out the skip which honestly is as good as a command line flag because I would bet most of the slow tests in projects with an `--enable-slow` flag _rarely_ get executed

**Comment by tanujkhattar** [comment]
> I would bet most of the slow tests in projects with an --enable-slow flag rarely get executed

The way it's handled in Cirq is that the slow tests get executed once as part of the daily CI while the rest of the tests get executed as part of every PR. I think this model is more robust and should be adopted. The reason slow tests exist is because we have found implementation bugs in the past which were not caught by small test cases, and having the daily CI running these slow tests will help us catch such bugs in case we change the implementation. The "run it by commenting out the skip" is not nearly as robust. Also, the infrastructure to add the daily CI and mark tests as slow is not terribly complicated and already exists in Cirq now; thanks to @pavoljuhas; so we can leverage the knowledge we've gained by implementing it in Cirq to simply adopt the same infrastructure here as well.

**Comment by mpharrigan** [comment]
Ok, let's merge this until the nightly CI is set up

**Review by tanujkhattar — APPROVED** [review]

**Comment by pavoljuhas** [comment]
@mpharrigan - I think it would be better to apply the skip mark at the test function definition rather than in its body.
If it is OK, I can add a commit to this PR to do so.

**Comment by mpharrigan** [comment]
These are parameterized tests using the "outer product" over the two parameter sets. Most of the parameter settings are not slow; just the cases identified by the `if ... skip` are slow.

**Comment by mpharrigan** [comment]
Got it, cool!

**Comment by pavoljuhas** [comment]
> Got it, cool!

Done in 66af330e5a91e517bbf7ea920027ebc74c3e0baf...d783c5186d0a90d70eb195918571e0064efb56f9

File: qualtran/bloqs/qrom_test.py [hunk 1]
```diff
@@ -27,10 +27,23 @@
 
 
 @pytest.mark.parametrize(
-    "data", [[[1, 2, 3, 4, 5]], [[1, 2, 3], [4, 5, 10]], [[1], [2], [3], [4], [5], [6]]]
+    "data,num_controls",
+    [
+        pytest.param(
```
Comment (mpharrigan) [inline qualtran/bloqs/qrom_test.py:32]: this is a really nice thing to know about, thanks @pavoljuhas !

File: qualtran/bloqs/qrom_test.py [hunk 2]
```diff
@@ -27,10 +27,23 @@
...
+        for idx, data in enumerate(
+            [[[1, 2, 3, 4, 5]], [[1, 2, 3], [4, 5, 10]], [[1], [2], [3], [4], [5], [6]]]
+        )
+        for num_controls in [0, 1, 2]
+    ],
 )
-@pytest.mark.parametrize("num_controls", [0, 1, 2])
 def test_qrom_1d(data, num_controls):
+    if num_controls == 2 and len(data) == 6:
+        pytest.skip("slow")
```
Comment (mpharrigan) [inline qualtran/bloqs/qrom_test.py:46]: I think this should be removed now, right?

**Comment by mpharrigan** [comment]
The problem is we don't actually use or respect the slow marker

File: qualtran/bloqs/qrom_test.py (hunk 2 above)
Comment (mpharrigan) [reply qualtran/bloqs/qrom_test.py:46]: oh I see two of them you skip in the decorator and this one you mark as slow and skip in the body. Is this to show how to do it when we set up a "slow" marker?

**Comment by pavoljuhas** [comment]
> The problem is we don't actually use or respect the slow marker

Yes, thank you for catching that.  I overlooked the mark after a copy-paste from Cirq.

We can setup a `slow` marker with its own test skipping logic like in Cirq, but that is a thing for a next PR.

**Comment by mpharrigan** [comment]
I opened https://github.com/quantumlib/Qualtran/issues/491 to track

**Pull request merged**